rust-runtime/smithy-json/src/serialize.rs

```diff
@@ -394,18 +394,15 @@ mod tests {
         assert_eq!("10000000000.0", format_test_number(Number::Float(1e10)));
         assert_eq!("-1.2", format_test_number(Number::Float(-1.2)));
 
-        // JSON doesn't support NaN, Infinity, or -Infinity, so we're matching
+        // Smithy has specific behavior for infinity & NaN
```

Comment:
given that these are _smithy_ specific libraries I think it's pretty reasonable. The behavior only augments the `expect_or` helpers instead of the raw token stream. We _could_ consider extracting those somewhere protocol specific

Reply:
but I think for now, a JSON library that "does the right thing" is probably the best option